authors/apps/authentication/serializers.py

```diff
@@ -25,6 +29,41 @@ class Meta:
         # or response, including fields specified explicitly above.
         fields = ['email', 'username', 'password']
 
+    def validate_email(self, value):
+        user_email_db = User.objects.filter(email=value)
+        if user_email_db.exists():
+            raise serializers.ValidationError("Email already in use")
+        return value
+
+    def validate_password(self, value):
+        if len(value) < 8:
+            raise serializers.ValidationError(
+                "Weak password. Password should be atleast 8 characters long")
+        elif len(value) > 25:
+            raise serializers.ValidationError(
+                "Password can not be more than 25 caharacters")
+        elif not re.search(r'[0-9]', value):
+            raise serializers.ValidationError(
+                "Weak password. Include atleast one integer")
+        elif value.isupper() or value.islower():
+            raise serializers.ValidationError(
+                "Password should contain both upper and lower cases")
+        elif value.isdigit():
+            raise serializers.ValidationError(
+                "Password can not contain only integers")
+        return value
+
+    def validate_username(self, value):
+        username_db = User.objects.filter(username=value)
+        if username_db.exists():
+            raise serializers.ValidationError("Username already in use")
+        elif len(value) <= 2 or len(value) > 25:
+            raise serializers.ValidationError(
```

Comment:
The same goes for a username, I think setting unique to true in the model when defining the fields guarantees it will always be unique, in times when you want to add more validation constraints, adding custom  validators would be more practicle  than rewriting the whole thing.  [You can Read more  here](https://www.django-rest-framework.org/api-guide/validators/#writing-custom-validators)